src/studio-home/tabs-section/libraries-v2-tab/libraries-v2-filters/index.jsx

```diff
@@ -0,0 +1,123 @@
+import { useState, useCallback, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import { SearchField } from '@openedx/paragon';
+import { debounce } from 'lodash';
+
+import { LoadingSpinner } from '../../../../generic/Loading';
+import LibrariesV2OrderFilterMenu from './libraries-v2-order-filter-menu';
+
+/* regex to check if a string has only whitespace
+  example "    "
+*/
+const regexOnlyWhiteSpaces = /^\s+$/;
+
+const LibrariesV2Filters = ({
+  isLoading,
+  isFiltered,
+  setIsFiltered,
+  setFilterParams,
+  setCurrentPage,
+}) => {
+  const [search, setSearch] = useState('');
+  const [order, setOrder] = useState('title');
+
+  // Reset search & order when filters cleared
+  useEffect(() => {
+    if (!isFiltered) {
+      setSearch('');
+      setOrder('title');
+    }
+  }, [isFiltered, setSearch, setOrder]);
+
+  const getOrderFromFilterType = (filterType) => {
+    const orders = {
+      azLibrariesV2: 'title',
+      zaLibrariesV2: '-title',
+      newestLibrariesV2: '-created',
+      oldestLibrariesV2: 'created',
+    };
+
+    // Default to 'A-Z` if invalid filtertype
+    return orders[filterType] || 'title';
+  };
+
+  const getFilterTypeData = (baseFilters) => ({
+    azLibrariesV2: { ...baseFilters, order: 'title' },
+    zaLibrariesV2: { ...baseFilters, order: '-title' },
+    newestLibrariesV2: { ...baseFilters, order: '-created' },
+    oldestLibrariesV2: { ...baseFilters, order: 'created' },
+  });
+
+  const handleMenuFilterItemSelected = (filterType) => {
+    setOrder(getOrderFromFilterType(filterType));
+    setIsFiltered(true);
+
+    const baseFilters = {
+      search,
+      order,
+    };
+
+    const filterParams = getFilterTypeData(baseFilters);
+    const filterParamsFormat = filterParams[filterType] || baseFilters;
+
+    setFilterParams(filterParamsFormat);
+    setCurrentPage(1);
+  };
+
+  const handleSearchLibrariesV2 = (searchValueDebounced) => {
+    const valueFormatted = searchValueDebounced.trim();
+    const filterParams = {
+      search: valueFormatted.length > 0 ? valueFormatted : undefined,
+      order,
+    };
+    const hasOnlySpaces = regexOnlyWhiteSpaces.test(searchValueDebounced);
+    if (valueFormatted !== search && !hasOnlySpaces) {
+      setIsFiltered(true);
+      setSearch(valueFormatted);
+      setFilterParams(filterParams);
+      setCurrentPage(1);
+    }
+  };
+
+  const handleSearchLibrariesV2Debounced = useCallback(
+    debounce((value) => handleSearchLibrariesV2(value), 400),
```

Comment:
Not strictly necessary, just thought it might help with the fast changes on the page when typing. I've removed it though to keep it consistent with the `search-modal`.